### PR TITLE
Filter checkins data based on their transmission risk level. mapping

### PR DESCRIPTION
**Pull request by EugenM-SAP**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure `mvn -P integration-tests clean verify` runs for the whole project and, if you touched any code for a service in the `services` folder, ensure it can be run with `spring-boot:run`

## Description

Any check-in with a transmissionRiskLevel that maps to a Transmission Risk Value (as per app-config) of 0 (or 0.0) shall be excluded from further processing (i.e. filtered out). This implies that the risk calculation paramters / trl mapping config added to 1.15 v2 gets moved to a common place that can be shared between Distribution and Submission service.

### CODE REVIEW

File: common/persistence/src/main/resources/trl-value-mapping-1.15.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+transmission-risk-value-mapping:
+  -
+    transmission-risk-level: 1
+    transmission-risk-value: 0
+  -
+    transmission-risk-level: 2
+    transmission-risk-value: 0
```
Comment (hilmarf) [inline common/persistence/src/main/resources/trl-value-mapping-1.15.yaml:7]: those will be then the ones, that will be filtered out during `submission`, right?

File: services/submission/src/main/java/app/coronawarn/server/services/submission/checkins/EventCheckinDataFilter.java [hunk 2]
```diff
@@ -61,4 +68,11 @@ private boolean filterOutFutureCheckins(CheckIn checkin) {
   private boolean filterByValidSignature(CheckIn checkin) {
     return traceLocationSignatureVerifier.verify(checkin.getSignedLocation());
   }
+
+  private boolean mapsTo(Integer valueToCheck, Double target) {
+    Optional<TransmissionRiskValueMapping> foundTarget =
+        trlValueMappingProvider.getTransmissionRiskValueMapping().stream()
+            .filter(m -> m.getTransmissionRiskLevel().equals(valueToCheck)).findAny();
+    return foundTarget.isPresent() && foundTarget.get().getTransmissionRiskValue().equals(target);
```
Comment (hilmarf) [inline services/submission/src/main/java/app/coronawarn/server/services/submission/checkins/EventCheckinDataFilter.java:76]: @EugenM-SAP / @mlenkeit - what about the ones, that are NOT `present`?

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/config/PreDistributionTrlValueMappingProvider.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import java.util.stream.Collectors;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+
+@Configuration
+@ConfigurationProperties()
+@PropertySource(value = "classpath:trl-value-mapping-1.15.yaml",
+    factory = YamlPropertySourceFactory.class)
+public class PreDistributionTrlValueMappingProvider {
```
Comment (hilmarf) [inline common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/config/PreDistributionTrlValueMappingProvider.java:14]: `PreDistributionTrlValueMappingProvider` name is a bit misleading, when we use it during the submission process, right?

File: common/persistence/src/main/resources/trl-value-mapping-1.15.yaml (hunk 1 above)
Comment (EugenM-SAP) [reply common/persistence/src/main/resources/trl-value-mapping-1.15.yaml:7]: @hilmarf  yes

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/config/PreDistributionTrlValueMappingProvider.java (hunk 3 above)
Comment (EugenM-SAP) [reply common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/config/PreDistributionTrlValueMappingProvider.java:14]: I agree. But somehow that is the context where the mapping is used for. My idea was to put this explicit naming to force us to think if there is a design issue we are missing here. Together with Max we did discuss other aproaches where the configuration can not be shared, but to be honest, the most optimal way is to filter out checkins during submission based on this "distribution-related" configuration. 
We could change the naming or revisit this at a later point in time.

File: services/submission/src/main/java/app/coronawarn/server/services/submission/checkins/EventCheckinDataFilter.java (hunk 2 above)
Comment (EugenM-SAP) [reply services/submission/src/main/java/app/coronawarn/server/services/submission/checkins/EventCheckinDataFilter.java:76]: In my view, whatever does not meet the TRL mapping value, should be filtered out. Which is what happens above.

**Review by hilmarf — APPROVED** [review]

**Pull request merged**